hack/jenkins/README.md

```diff
@@ -0,0 +1,28 @@
+# Jenkins
+
+[Jenkins](http://jenkins-ci.org/) is a pluggable continuous
+integration system. The Google team is running a Jenkins server on a
+private GCE instance for the Kubernetes project in order to run longer
+integration tests, continuously, on different providers. Currently, we
+(Google) are only running Jenkins on our own providers (GCE and GKE)
+in different flavors.
+
+## General flow
+The flow of the Google Jenkins server:
+* Under the `kubernetes-build` job: Every 5 minutes, Jenkins polls for a batch of new commits, after which it runs the `build.sh` script (in this directory) on the latest tip. This results in build assets getting pushed to GCS and the `latest.txt` file in the `ci` bucket being updated. That job then triggers `kubernetes-e2e-*`.
+* On trigger, and every half hour (which effectively means all the time, unless we're failing cluster creation), e2e variants run, on the latest build assets in GCS:
+  * `kubernetes-e2e-gce`: Standard GCE e2e
+  * `kubernetes-e2e-gke`: GKE provider e2e, with head k8s and GKE creating clusters at its default version
+  * `kubernetes-e2e-gke-ci`: GKE provider e2e, with head k8s and GKE creating clusters at the head k8s version
+* Each named job may run in parallel with the others, i.e. `kubernetes-build` with `kubernetes-e2e-gce`, but `kubernetes-e2e-gce` will not run concurrently with itself.
```

Comment:
BTW, if you're just asking the question: No, you can only have 1 kubernetes-build and 1 kubernetes-e2e-gce test running at any given time today. Eventually for e2es I'd like that to be multiple e2es so that we can potentially kick them off as commits come in, but right now as things roll in, they just get batched up.